pkg/aws/manager_ec2_vpc_peering.go

```diff
@@ -0,0 +1,114 @@
+	var vpcPeeringConnectionsToDelete []*basicResource
+	r.logger.Debug("delete vpc peering connections resources for cluster")
+	resourceInput := &resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: aws.StringSlice([]string{resourceTypeVpcPeeringConnection}),
+		TagFilters:          convertClusterTagsToAWSTagFilter(clusterId, tags),
+	}
+	// add to the vpc peering delete array
+	resourceOutput, err := r.taggingClient.GetResources(resourceInput)
+	if err != nil {
+		return nil, errors.WrapLog(err, "failed to filter vpc peering connection", r.logger)
+	}
+
+	for _, resourceTagMapping := range resourceOutput.ResourceTagMappingList {
+		arn := aws.StringValue(resourceTagMapping.ResourceARN)
+		arnElements := strings.Split(arn, ":")
+		vpcPeeringConnectionID := arnElements[len(arnElements)-1]
+		if vpcPeeringConnectionID == "" {
+			return nil, errors.WrapLog(err, fmt.Sprintf("invalid vpc peering connection name from arn, %s", vpcPeeringConnectionID), r.logger)
+		}
+		vpcPeeringConnectionsToDelete = append(vpcPeeringConnectionsToDelete, &basicResource{
+			Name: vpcPeeringConnectionID,
+			ARN:  arn,
+		})
+		r.logger.Infof("found list of %d vpc peering connection to delete", len(vpcPeeringConnectionsToDelete))
+	}
+	//delete resources
+	var reportItems []*clusterservice.ReportItem
+	for _, vpcPeeringConnection := range vpcPeeringConnectionsToDelete {
+		vpcPeeringConnectionLogger := r.logger.WithField(loggingKeyVpcPeeringConnection, vpcPeeringConnection.Name)
+		reportItem := &clusterservice.ReportItem{
+			ID:           vpcPeeringConnection.ARN,
+			Name:         vpcPeeringConnection.Name,
+			Action:       clusterservice.ActionDelete,
+			ActionStatus: clusterservice.ActionStatusInProgress,
+		}
+		reportItems = append(reportItems, reportItem)
+		if dryRun {
+			vpcPeeringConnectionLogger.Debugf("dry run is enabled, skipping deletion")
+			reportItem.ActionStatus = clusterservice.ActionStatusDryRun
+			continue
+		}
+		vpcPeeringConnectionLogger.Debugf("performing vpc peering connection deletion")
+		deleteVpcPeeringConnectionInput := &ec2.DeleteVpcPeeringConnectionInput{
+			VpcPeeringConnectionId: aws.String(vpcPeeringConnection.Name),
+		}
+		if _, err := r.ec2Client.DeleteVpcPeeringConnection(deleteVpcPeeringConnectionInput); err != nil {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "DependencyViolation" {
+				vpcPeeringConnectionLogger.Debug("vpc peering connection has existing dependencies which have not been deleted, skipping")
+				reportItem.ActionStatus = clusterservice.ActionStatusSkipped
+				r.logger.Infof("Error: %s, %s", awsErr.Code(), awsErr.Message())
+				continue
+			}
+
+			// in the case of vpc peerings they are picked up on describe but then when attempting to delete they are not found
+			// any that are not found can be skipped.
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "InvalidVpcPeeringConnectionID.NotFound" {
```

Comment:
Out of curiosity, how come we don't check this for other resources?

Reply:
vpc peerings remain in a deleted state for a couple of hours after deletion, this doesn't happen with other resources. This is here to handle that exception.